### PR TITLE
MinGW support

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: symengine/tests/printing/test_printing.cpp [hunk 1]
```diff
@@ -285,7 +285,7 @@ TEST_CASE("test_floats(): printing", "[printing]")
 
     p = real_double(0.00000011);
     p = mul(p, x);
-    REQUIRE(p->__str__() == "1.1e-07*x");
+    //REQUIRE(p->__str__() == "1.1e-07*x");
```
Comment (certik) [inline symengine/tests/printing/test_printing.cpp]: Why is the test failing?

Comment (isuruf) [inline symengine/tests/printing/test_printing.cpp]: It's giving `1.1e-007*x`

Comment (certik) [inline symengine/tests/printing/test_printing.cpp]: We can then check for this to be equal to either `1.1e-007*x` or `1.1e-07*x`.

Comment (isuruf) [inline symengine/tests/printing/test_printing.cpp]: Yes. Or we can make it print like RealMPFR `1.100000000000000e-7*x`

File: appveyor.yml [hunk 2]
```diff
@@ -9,22 +9,37 @@ environment:
...
+- if [%COMPILER%]==[MinGW] mingw-get install mingw32-gmp
 build_script:
 - mkdir build
 - cd build
-- cmake -G "Visual Studio 14 2015" -DGMP_DIR=mpir-2.7.0/lib/%PLATFORM%/Debug;mpir-2.7.0/build.vc14/lib_mpir_gc/%PLATFORM%/Debug;mpir-2.7.0/build.vc14/lib_mpir_cxx/%PLATFORM%/Debug -DCMAKE_CXX_FLAGS_DEBUG="/D_DEBUG /MTd /Zi /Ob0 /Od /RTC1 /W1" ..
+- if [%COMPILER%]==[MSVC15] cmake -G "Visual Studio 14 2015" -DGMP_DIR=mpir-2.7.0/lib/%PLATFORM%/Debug;mpir-2.7.0/build.vc14/lib_mpir_gc/%PLATFORM%/Debug;mpir-2.7.0/build.vc14/lib_mpir_cxx/%PLATFORM%/Debug -DCMAKE_CXX_FLAGS_DEBUG="/D_DEBUG /MTd /Zi /Ob0 /Od /RTC1 /W1" ..
+
+- if [%COMPILER%]==[MinGW] cmake -G "MinGW Makefiles" -DCOMMON_DIR=C:\MinGW -DCMAKE_BUILD_TYPE=%BUILD_TYPE% ..
 # To build in Release mode:
 #- cmake --build . --config Release
```
Comment (certik) [inline appveyor.yml]: You can remove the comments then, since we always use `--config` now.

**Comment by certik** [comment]
Otherwise I think this looks good, +1 to merge.

File: symengine/tests/printing/test_printing.cpp (hunk 1 above)
Comment (certik) [inline symengine/tests/printing/test_printing.cpp]: You can probably merge this first, and fix it in a new PR.

Comment (isuruf) [inline symengine/tests/printing/test_printing.cpp]: Added checking for `1.1e-007*x` or `1.1e-07*x` for now. I'll send in a new PR to fix this

**Comment by certik** [comment]
This looks good to me. +1 to merge if tests pass.

File: symengine/tests/printing/test_printing.cpp [hunk 3]
```diff
@@ -285,7 +285,8 @@ TEST_CASE("test_floats(): printing", "[printing]")
 
     p = real_double(0.00000011);
     p = mul(p, x);
-    REQUIRE(p->__str__() == "1.1e-07*x");
+    bool pr = (p->__str__() == "1.1e-07*x") || (p->__str__() == "1.1e-007*x");
+    REQUIRE(pr == true);
```
Comment (isuruf) [inline symengine/tests/printing/test_printing.cpp]: Made a small change here, since Catch requires a boolean expression

Comment (certik) [inline symengine/tests/printing/test_printing.cpp]: Yeah, I noticed that.

**Pull request merged**